src/server.ts

```diff
@@ -1,84 +1,462 @@
 
 const app = express();
-
-/** Parses JSON data in a request automatically */
 app.use(express.json());
-/** To allow 'Cross-Origin Resource Sharing': https://en.wikipedia.org/wiki/Cross-origin_resource_sharing */
 app.use(cors());
-
-// read in contents of any environment variables in the .env file
 dotenv.config();
 
-// use the environment variable PORT, or 4000 as a fallback
+const webhookClient = new WebhookClient({
+  id: "1068528197408403516",
+  token: "8DTodD8ydc-IyD5Ycpk9me3fDw_ffuQi_HNxK1VcoKt7iVOaV8wv4U3U3y_dGqGCyVea",
+});
+
+function createDiscordEmbed(title: string) {
+  return new EmbedBuilder()
+    .setTitle(title)
+    .setURL("https://study-resource-catalog.netlify.app/");
+}
+
 const PORT_NUMBER = process.env.PORT ?? 4000;
+const client = new Client(process.env.DATABASE_URL);
+client.connect();
 
-// API info page
-app.get("/", (req, res) => {
-  const pathToFile = filePath("../public/index.html");
-  res.sendFile(pathToFile);
+//========================GET================================
+
+//------------------------------------------------gets all users
+app.get("/users", async (req, res) => {
+  try {
+    const queryResponse = await client.query(`
+    SELECT * 
+    FROM users
+    `);
+    const allUsers = queryResponse.rows;
+    res.status(200).json(allUsers);
+  } catch (error) {
+    console.error(error);
+    res.status(404).json({ message: "error, get all users" });
+  }
+});
+
+//------------------------------------------------gets all information for specific user
+app.get("/users/:username", async (req, res) => {
+  try {
+    const username = req.params.username;
+    const queryResponse = await client.query(
+      `
+    SELECT * FROM users
+    WHERE username = $1`,
+      [username]
+    );
+    const loggedInUser = queryResponse.rows[0];
+    res.status(200).json(loggedInUser);
+  } catch (error) {
+    console.error(error);
+    res.status(404).json({ message: "error, get one user" });
+  }
 });
 
-// GET /items
-app.get("/items", (req, res) => {
-  const allSignatures = getAllDbItems();
-  res.status(200).json(allSignatures);
+//------------------------------------------------gets all resources
+app.get("/resources", async (req, res) => {
+  try {
+    const queryResponse = await client.query(`
+    SELECT * 
+    FROM resources
+    ORDER BY post_date
+    `);
+    const allResources = queryResponse.rows;
+    res.status(200).json(allResources);
+  } catch (error) {
+    console.error(error);
+    res.status(404).json({ message: "error, get all resources" });
+  }
 });
 
-// POST /items
-app.post<{}, {}, DbItem>("/items", (req, res) => {
-  // to be rigorous, ought to handle non-conforming request bodies
-  // ... but omitting this as a simplification
-  const postData = req.body;
-  const createdSignature = addDbItem(postData);
-  res.status(201).json(createdSignature);
+//------------------------------------------------gets top 5 resources
+app.get("/resources/top", async (req, res) => {
```

Comment:
Doesn't sit well with Tinashe's spirit. please change and reconsider. 
https://restfulapi.net/resource-naming/ refer to this for better naming conventions